From 89087c89c24af484348f7b99f55178eb4fb30af6 Mon Sep 17 00:00:00 2001
From: Fiona Klute <fiona.klute@gmx.de>
Date: Sat, 9 Mar 2024 13:24:02 +0100
Subject: [PATCH 41/48] 8723cs: Remove unused code that causes warnings

"buf_star" being uninitialized caused a warning every time the header
was included.

Signed-off-by: Fiona Klute <fiona.klute@gmx.de>
---
 drivers/staging/rtl8723cs/include/rtw_recv.h | 40 --------------------
 1 file changed, 40 deletions(-)

diff --git a/drivers/staging/rtl8723cs/include/rtw_recv.h b/drivers/staging/rtl8723cs/include/rtw_recv.h
index a2574f755434..229c17e83979 100644
--- a/drivers/staging/rtl8723cs/include/rtw_recv.h
+++ b/drivers/staging/rtl8723cs/include/rtw_recv.h
@@ -799,46 +799,6 @@ __inline static u8 *recvframe_pull_tail(union recv_frame *precvframe, sint sz)
 
 }
 
-__inline static union recv_frame *rxmem_to_recvframe(u8 *rxmem)
-{
-	/* due to the design of 2048 bytes alignment of recv_frame, we can reference the union recv_frame */
-	/* from any given member of recv_frame. */
-	/* rxmem indicates the any member/address in recv_frame */
-
-	return (union recv_frame *)(((SIZE_PTR)rxmem >> RXFRAME_ALIGN) << RXFRAME_ALIGN);
-
-}
-
-__inline static union recv_frame *pkt_to_recvframe(_pkt *pkt)
-{
-
-	u8 *buf_star;
-	union recv_frame *precv_frame;
-	precv_frame = rxmem_to_recvframe((unsigned char *)buf_star);
-
-	return precv_frame;
-}
-
-__inline static u8 *pkt_to_recvmem(_pkt *pkt)
-{
-	/* return the rx_head */
-
-	union recv_frame *precv_frame = pkt_to_recvframe(pkt);
-
-	return	precv_frame->u.hdr.rx_head;
-
-}
-
-__inline static u8 *pkt_to_recvdata(_pkt *pkt)
-{
-	/* return the rx_data */
-
-	union recv_frame *precv_frame = pkt_to_recvframe(pkt);
-
-	return	precv_frame->u.hdr.rx_data;
-
-}
-
 
 __inline static sint get_recvframe_len(union recv_frame *precvframe)
 {
-- 
2.43.0

